server/sonar-server/src/main/java/org/sonar/server/ce/ws/TaskFormatter.java

```diff
@@ -55,23 +57,28 @@ public TaskFormatter(DbClient dbClient, System2 system2) {
   }
 
   public List<WsCe.Task> formatQueue(DbSession dbSession, List<CeQueueDto> dtos) {
-    ComponentDtoCache cache = new ComponentDtoCache(dbSession, ceQueueDtoToComponentUuids(dtos));
+    ComponentDtoCache cache = ComponentDtoCache.forQueueDtos(dbClient, dbSession, dtos);
     return dtos.stream().map(input -> formatQueue(input, cache)).collect(Collectors.toList());
   }
 
   public WsCe.Task formatQueue(DbSession dbSession, CeQueueDto dto) {
-    return formatQueue(dto, new ComponentDtoCache(dbSession, dto.getComponentUuid()));
+    return formatQueue(dto, ComponentDtoCache.forUuid(dbClient, dbSession, dto.getComponentUuid()));
   }
 
   private WsCe.Task formatQueue(CeQueueDto dto, ComponentDtoCache componentDtoCache) {
     WsCe.Task.Builder builder = WsCe.Task.newBuilder();
+    String organizationKey = componentDtoCache.getOrganizationKey(dto.getComponentUuid());
+    // FIXME organization field should be set from the CeQueueDto rather than from the ComponentDto
```

Comment:
it is not enforced for a task to be associated to a component, therefor it may not have an organization.
In addition, a component may be deleted and a task still be there.